ocs_ci/ocs/ui/page_objects/bucket_class_tab.py

```diff
@@ -0,0 +1,113 @@
+import ipaddress
+import random
+import re
+import string
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.ui.base_ui import logger
+from ocs_ci.ocs.ui.page_objects.data_foundation_tabs_common import (
+    DataFoundationDefaultTab,
+    CreateResourceForm,
+    DataFoundationTabBar,
+)
+
+
+class BucketClassTab(DataFoundationDefaultTab, CreateResourceForm):
+    def __init__(self):
+        DataFoundationTabBar.__init__(self)
+        CreateResourceForm.__init__(self)
+        self.rules = {
+            constants.UI_INPUT_RULES_BUCKET_CLASS["rule1"]: self._check_3_63_char_rule,
+            constants.UI_INPUT_RULES_BUCKET_CLASS[
+                "rule2"
+            ]: self._check_start_end_char_rule,
+            constants.UI_INPUT_RULES_BUCKET_CLASS[
+                "rule3"
+            ]: self._check_only_lower_case_numbers_periods_hyphens_rule,
+            constants.UI_INPUT_RULES_BUCKET_CLASS[
+                "rule4"
+            ]: self._check_no_ip_address_rule,
+            constants.UI_INPUT_RULES_BUCKET_CLASS[
+                "rule5"
+            ]: self._check_bucketclass_name_not_used_before_rule,
+        }
+        self.name_input_loc = self.bucketclass["bucketclass_name"]
+
+    def _check_3_63_char_rule(self, rule_exp) -> bool:
+        """
+        Check if the input text length between 3 and 63 characters only can be used.
+
+        Args:
+            rule_exp (str): the rule requested to be checked. rule_exp text should match the text from validation popup
+
+        Returns:
+            bool: True if the input text length not violated, False otherwise.
+        """
+        logger.info(f"checking the input rule '{rule_exp}'")
+        numbers = re.findall(r"\d+", rule_exp)
+        min_len, max_len = map(int, numbers)
+        params_list = [
+            (rule_exp, min_len - 1, self.status_error),
+            (rule_exp, min_len, self.status_success),
+            (rule_exp, min_len + 1, self.status_success),
+            (rule_exp, max_len - 1, self.status_success),
+            (rule_exp, max_len, self.status_success),
+            (rule_exp, max_len + 1, self.status_error),
+        ]
+
+        return all(self._check_input_text_length(*params) for params in params_list)
+
+    def _check_no_ip_address_rule(self, rule_exp) -> bool:
+        """
+        Check if the input does not contain a valid IPv4 address.
+
+        This function generates a random IPv4 address and a random string that is not an IP address.
+
+        Args:
+            rule_exp (str): the rule requested to be checked. rule_exp text should match the text from validation popup
+
+        Returns:
+            bool: True if the rule is satisfied for the random string that is not an IP address, False otherwise.
+        """
+
+        def _generate_ipv4_address_str():
```

Comment:
not very crucial but maybe in a follow up PR - this doesn't seem to belong to UI functionalities. Suggest moving it to a general utils file